Src/main.c

```diff
@@ -273,80 +235,122 @@ static void MX_ICACHE_Init(void)
 /* USER CODE BEGIN 4 */
 
 /* USER CODE END 4 */
+/**
+  * @brief EXTI line detection callbacks
+  * @param GPIO_Pin: Specifies the pins connected EXTI line
+  * @retval None
+  */
+static void EXTI13_IRQHandler_Config(void)
+{
+  GPIO_InitTypeDef   GPIO_InitStructure;
+
+
+  /* Enable GPIOC clock */
+  __HAL_RCC_GPIOC_CLK_ENABLE();
+
+  /* Configure PC.13 pin as input floating */
+  GPIO_InitStructure.Mode = GPIO_MODE_IT_RISING_FALLING;
+
+
+  GPIO_InitStructure.Pull = GPIO_NOPULL;
+  GPIO_InitStructure.Pin = BUTTON_USER_PIN;
+  HAL_GPIO_Init(GPIOC, &GPIO_InitStructure);
+
+
+  /* Enable and set line 13 Interrupt to the lowest priority */
+  HAL_NVIC_SetPriority(EXTI13_IRQn, 2, 0);
+  HAL_NVIC_EnableIRQ(EXTI13_IRQn);
+}
+
 
-/* USER CODE BEGIN Header_LED_Thread1 */
 /**
-  * @brief  Function implementing the THREAD1 thread.
-  * @param  argument: Not used
+  * @brief Task1 handler and creating a autoreload timer
+  * @param None
   * @retval None
   */
-/* USER CODE END Header_LED_Thread1 */
-void LED_Thread1(void *argument)
+
+void  vPeriodicTask(void *argument)
 {
-  /* USER CODE BEGIN 5 */
-   struct student *t;
-  /* USER CODE BEGIN 5 */
-  //uint32_t count = 0;
-	//UBaseType_t uxPriority;
-  t= (struct student *) argument;
-
-  /*Query the priority at which this task is running - passing in NULL means "return the calling task’s priority". */
-   //uxPriority = uxTaskPriorityGet( NULL );
-
-	  while(1)
-    {
-      BSP_LED_Toggle(LED9);
-      printf("Task1 is running\n");
-      printf("Num :%d\n",t->num);
-      printf("Name :%s\n",t->name);
-      printf("Age :%d\n\n",t->age);
-
- /* Setting the Task 2 priority above the Task 1 priority will cause Task 2 to immediately start running */
-    //  printf( "About to raise the Task 2 priority\r\n" );
-    //  vTaskPrioritySet(task2_handle, ( uxPriority + 1 ) );
-      osDelay(500);
-    }
-
-
-  /* USER CODE END 5 */
+	TimerHandle_t xAutoReloadTimer;
+	//BaseType_t xTimer1Started;
+	printf("Task1\n");
+	xAutoReloadTimer = xTimerCreate( "AutoReload", mainAUTO_RELOAD_TIMER_PERIOD, pdTRUE,0,  prvAutoReloadTimerCallback );
```

Comment:
Is there any specific reason to create and start timer from Task?
Can we remove task or not?
Think of memory utilization and execution time perspective first while developing MCU application.